docs/integration_setup_failures.md

```diff
@@ -0,0 +1,86 @@
+---
+title: "Handling Setup Failures"
+---
+
+Your integration may not be able to be set up for a variety of reasons. The most common cases are because the device or service is offline or the credentials are no longer valid. Your integration must retry setup so it can recover as soon as reasonably possible when the device or service is back online without the user having to restart Home Assistant.
+
+## Handling offline or unavailable devices and services
+
+### Integrations using `async_setup_entry`
+
+Raise the `ConfigEntryNotReady` exception from `async_setup_entry` in the integrations `__init__.py`, and Home Assistant will automatically take care of retrying set up later. To avoid doubt, raising `ConfigEntryNotReady` in a platform's `async_setup_entry` is ineffective because it is too late to be caught by the config entry setup.
```

Comment:
```suggestion
Raise the `ConfigEntryNotReady` exception from `async_setup_entry` in the integration's `__init__.py`, and Home Assistant will automatically take care of retrying set up later. To avoid doubt, raising `ConfigEntryNotReady` in a platform's `async_setup_entry` is ineffective because it is too late to be caught by the config entry setup.
```